codice/ddf
# DDF-5381 Allow admin layout to override visualizations

#### What does this PR do?
Allow admin layout to support custom visualizations

#### Who is reviewing it? 

@jordanwilking 
@stevenmalmgren 

#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@djblue 
-->

#### How should this be tested?
Open `$DDF_DIST/ddf-2.19.0-SNAPSHOT/etc/org.codice.ddf.catalog.ui.config` and add a config for visualizations like this:

```
"[{\"name\":\"openlayers\",\"title\":\"2D Map\",\"icon\":\"map\"},{\"name\":\"cesium\",\"title\":\"3D Map\",\"icon\":\"globe\"},{\"name\":\"inspector\",\"title\":\"Inspector\",\"icon\":\"info\"},{\"name\":\"histogram\",\"title\":\"Histogram\",\"icon\":\"bar-chart\"},{\"name\":\"table\",\"title\":\"Table\",\"icon\":\"table\"},{\"name\":\"timeline\",\"title\":\"Timeline\",\"icon\":\"hourglass\"},{\"name\":\"results\",\"title\":\"Results\",\"icon\":\"list\"}]"
```

The list structure is `component,componentName,font-awesome4.7.0-icon-name`

Then verify that you can choose timeline or results custom visualizations in the admin ui > search UI > layout tab

#### Any background context you want to provide?

#### What are the relevant tickets?
Fixes: #5381

#### Screenshots

![image](https://user-images.githubusercontent.com/7236299/65698984-3e511a80-e032-11e9-92b3-ed5fbff2fa71.png)
(Screenshot shows list in a force-open state as admin-layout dev server has some issues with material ui, it will look like normal when you build the distribution)


#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
